be2-scala/src/main/scala/ch/epfl/pop/model/objects/Uint53.scala

```diff
@@ -0,0 +1,38 @@
+package ch.epfl.pop.model.objects
+
+
+class ArithmeticOverflowError(msg: String) extends Error(msg)
+
+
+/**
+ * Operations on 53-bits non-negative integers.
+ *
+ * These values can be handled without loss of precision in IEEE754 64-bit
+ * floating numbers, and thus in JSON.
+ */
+object Uint53 {
+  /** Internal representation in this implementation */
+  type Repr = Long
+
+  /** Smallest allowed value */
+  final val MinValue: Uint53 = 0
+
+  /** Largest allowed value */
+  final val MaxValue: Uint53 = 0x1FFFFFFFFFFFFFL
+
+  /** Whether the value is in-range */
+  def inRange(v: Uint53) = v >= MinValue && v <= MaxValue
+
+  /** Compute the sum of many Uint53 values, or indicate an overflow if
+   *  the sum is too large. */
+  def safeSum(seq: IterableOnce[Uint53]): Either[ArithmeticOverflowError, Uint53] = {
```

Comment:
I would document here that the Left/Right convention is the general one and not the "wrong" one used in the project, just to avoid confusion